src/App.tsx

```diff
@@ -1,24 +1,248 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFooter } from './components/TodoFooter/TodoFooter';
+import './App.scss';
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+import { TodoNotification }
+  from './components/TodoNotification/TodoNotification';
+import {
+  createTodo,
+  getTodos, removeTodo, updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterOptions } from './types/FilterOptions';
+import { filterTodos } from './Helpers';
+import { ErrorMessages } from './types/ErrorMessages';
+import { UpdateTodoArgs } from './types/UpdateTodoArgs';
 
-const USER_ID = 0;
+const USER_ID = 6795;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterOption, setFilterOption] = useState(FilterOptions.ALL);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessages | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [updatingTodosId, setUpdatingTodosId] = useState<number[]>([]);
+  const [isAdding, setIsAdding] = useState<boolean>(false);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorMessages.LoadError);
+      });
+  }, []);
+
+  const visibleTodos = useMemo(() => filterTodos(todos, filterOption),
+    [filterOption, todos]);
+
+  const activeVisibleTodos = useMemo(
+    () => filterTodos(todos, FilterOptions.ACTIVE),
+    [todos],
+  );
+  const completedVisibleTodos = useMemo(
+    () => filterTodos(todos, FilterOptions.COMPLETED),
+    [todos],
+  );
+  const isTodosPresent = todos.length > 0;
+  const isAllTodosCompleted = todos.every(todo => todo.completed);
+
+  const clearNewTodoTitle = () => (
+    setNewTodoTitle('')
+  );
+
+  const addNewTodo = async (title: string) => {
+    if (!title.trim()) {
+      setErrorMessage(ErrorMessages.TitleError);
+
+      return;
+    }
+
+    try {
+      const newTodoPayload = {
+        completed: false,
+        title,
+        userId: USER_ID,
+      };
+
+      setTempTodo({
+        id: 0,
+        ...newTodoPayload,
+      });
+
+      const newTodo = await createTodo(newTodoPayload);
+
+      clearNewTodoTitle();
```

Comment:
This function needs to add new todo only.
Clear the field after this function on submitting form

```suggestion
```